### DISCUSSION TITLE
Combine a split version declaration in Maven

### DISCUSSION THREAD

**Discussion by luk-lo** [renovatebot/renovate#21722] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.54.0

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

3.6.9

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

I have a version declaration in Maven like this:
```xml
<properties>
  <framework-version>2.7.8</framework-version>
  <patched-framework-version>${framework-version}-patch05</patched-framework-version>
</properties>

<dependencies>
  <dependency>
    <groupId>com.example</groupId>
    <artifactId>patched-framework-dependencies</artifactId>
    <version>${patched-framework-version}</version>
  </dependency>
</dependencies>

<plugins>
  <plugin>
    <groupId>com.example</groupId>
    <artifactId>framework-maven-plugin</artifactId>
    <version>${framework-version}</version>
  </plugin>
</plugins>
```

I would like to configure Renovate in a way that makes it upgrade `framework-version` only when a corresponding update of `patched-framework-dependencies` exist, e.g. update to `2.7.9` only when `2.7.9-patch01` is released.

### What you tried so far.

I can almost always make Renovate do what I want with a `regexManager`, so I tried this here too.
I managed to combine the `currentVersion` of `patched-framework` like this:
```json
      {
          "fileMatch": ["pom\\.xml$"],
          "matchStrings": ["<framework-version>(?<frameworkVersion>.*?)<framework-version>",
              "<patched-framework-version>\\${framework-version}(?<patchVersion>.*?)<\/framework-version>"],
          "matchStringsStrategy": "combination",
          "depNameTemplate": "com.example:patched-framework-dependencies",
          "datasourceTemplate": "maven",
          "currentVersionTemplate": "{{{frameworkVersion}}}{{{patchVersion}}}",
          "versioningTemplate": "maven"
      }
```
With this Renovate finds a new version `2.7.9-patch-01` but since that string is not in the file continuously it can't put together an update. Any ideas on how I could solve this? If I separately declare the versions it boils down to making the plugin take the version from the other dependency with the `-patch-*` cut off but can Renovate do this? I have experimented with setting `autoReplaceStringTemplate` and `extractVersionTemplate` but I think they are not the right tools here.

**Comment by viceice** [comment, marked as answer]
no, not possible to update splitted versions
